.github/workflows/ci.yml

```diff
@@ -35,20 +35,18 @@ jobs:
                   coverage: "none"
                   php-version: ${{ matrix.php-version }}
                   ini-values: memory_limit=-1
+                  tools: phpunit
```

Comment:
I would suggest to use a dev requirement instead, which would have 2 benefits:

- the CI would use a known PHPUnit version instead of whatever `setup-php` decides to use (the current setup is essentially a [death star version constraint](https://thephp.cc/articles/the-death-star-version-constraint)
- it will make it easy for contributors to use the appropriate PHPUnit version locally

Reply:
the issue is not that it is a secret. The issue is that it can upgrade to a new major version of PHPUnit without any work done in Twig (making the CI fail on an unrelated contribution in Twig after such upgrade of PHPUnit is released externally).
As explained in the blog post I linked, upgrading to a new major version must be a conscious choice (made by the Twig team, not by the setup-php maintainer).

Reply:
and Psalm is not installed through require-dev because Psalm is not used at all (if you thought about the Symfony repo, Psalm is used in a very special way there, so we explicitly don't want contributors to have it locally)

Reply:
> upgrading to a new major version must be a conscious choice

That makes total sense. I'm pinning PHPUnit to 11.3 now.